cli/azd/pkg/account/manager.go

```diff
@@ -0,0 +1,230 @@
+		}
+
+		azdConfig.DefaultSubscription = defaultSubscription
+	}
+
+	if azdConfig.DefaultLocation == nil {
+		defaultLocation, err := m.getDefaultLocation(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving default location from Azure CLI")
+		}
+
+		azdConfig.DefaultLocation = defaultLocation
+	}
+
+	return azdConfig, nil
+}
+
+// Gets the available Azure subscriptions for the current logged in account.
+func (m *Manager) GetSubscriptions(ctx context.Context) ([]azcli.AzCliSubscriptionInfo, error) {
+	defaultSubscription, err := m.getDefaultSubscription(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	accounts, err := m.azCli.ListAccounts(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed listing azure subscriptions: %w", err)
+	}
+
+	if defaultSubscription == nil {
+		return accounts, nil
+	}
+
+	// If default subscription is set, set it in the results
+	results := []azcli.AzCliSubscriptionInfo{}
+	for _, sub := range accounts {
+		if sub.Id == defaultSubscription.Id {
+			sub.IsDefault = true
+		}
+		results = append(results, sub)
+	}
+
+	return results, nil
+}
+
+// Gets the available Azure locations for the default Azure subscription.
+func (m *Manager) GetLocations(ctx context.Context) ([]azcli.AzCliLocation, error) {
+	defaultSubscription, err := m.getDefaultSubscription(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	locations, err := m.azCli.ListAccountLocations(ctx, defaultSubscription.Id)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving Azure location for account '%s': %w", defaultSubscription.Id, err)
+	}
+
+	return locations, nil
+}
+
+// Sets the default Azure subscription for the current logged in account.
+func (m *Manager) SetDefaultSubscription(ctx context.Context, subscriptionId string) (*config.Subscription, error) {
+	azdConfig, err := config.Load()
+	if err != nil {
+		azdConfig = &config.Config{}
+	}
+
+	subscription, err := m.azCli.GetAccount(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting account for id '%s'", subscriptionId)
+	}
+
+	azdConfig.DefaultSubscription = &config.Subscription{
+		Id:       subscription.Id,
+		Name:     subscription.Name,
+		TenantId: subscription.TenantId,
+	}
+
+	err = azdConfig.Save()
+	if err != nil {
+		return nil, fmt.Errorf("failed saving AZD configuration: %w", err)
+	}
+
+	return azdConfig.DefaultSubscription, nil
+}
+
+// Sets the default Azure location for the current logged in account.
+func (m *Manager) SetDefaultLocation(ctx context.Context, location string) (*config.Location, error) {
+	azdConfig, err := config.Load()
+	if err != nil {
+		azdConfig = &config.Config{}
+	}
+
+	locations, err := m.GetLocations(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving locations: %w", err)
+	}
+
+	index := slices.IndexFunc(locations, func(l azcli.AzCliLocation) bool {
+		return l.Name == location
+	})
+
+	if index < 0 {
+		return nil, fmt.Errorf("location '%s' is not valid", location)
+	}
```

Comment:
is it worth it to do this?

I see the benefit to folks entering typos, with the tradeoff of the command taking longer time to execute.

But, if I set a default location for `azd` and then I run `az logout` and `az login` to a different account, the defaults for location/subscription-name/etc might be still persisted in the home user json but not valid anymore...

Reply:
I think it's important that we validate the subscription/location at time of setting to protect against typos, etc. 
Agree that today this wouldn't support logging out & in as a new account from `az`.  But as we implement our own `azd login` we could then control what happens in our own config file in this use case.